src/bundle/Controller/VersionController.php

```diff
@@ -7,40 +7,34 @@
  */
 namespace EzSystems\HybridPlatformUiBundle\Controller;
 
-use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use eZ\Publish\API\Repository\Values\Content\Content;
 use EzSystems\HybridPlatformUi\Form\UiFormFactory;
 use EzSystems\HybridPlatformUi\Repository\UiVersionService;
-use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Routing\RouterInterface;
 
-class VersionController extends Controller
+class VersionController extends TabController
 {
     public function draftActionsAction(
-        $contentId,
+        Content $content,
         Request $request,
         UiVersionService $versionService,
-        UiFormFactory $formFactory,
-        RouterInterface $router
+        UiFormFactory $formFactory
     ) {
         $draftActionsForm = $formFactory->createVersionsDraftActionForm();
         $draftActionsForm->handleRequest($request);
 
+        $redirectLocationId = $request->query->get('redirectLocationId', $content->contentInfo->mainLocationId);
```

Comment:
it needs to be added to a later PR to fix the urls, they should all pass location but we need a fallback